test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,675 @@
+    {
+        /* Track reverse_collator value as it is required for the custom comparator. */
+        reverse_collator_enabled = _config->get_bool(REVERSE_COLLATOR);
+    }
+
+    bool
+    custom_lexicographical_compare(
+      const std::string &first_key, const std::string &second_key, bool inclusive)
+    {
+        if (this->reverse_collator_enabled)
+            if (!inclusive)
+                return first_key.compare(second_key) > 0;
+            else
+                return first_key.compare(second_key) >= 0;
+        else if (!inclusive)
+            return first_key.compare(second_key) < 0;
+        else
+            return first_key.compare(second_key) <= 0;
+    }
+
+    void
+    cursor_traversal(scoped_cursor &range_cursor, scoped_cursor &normal_cursor,
+      const bound &lower_bound, const bound &upper_bound, bool next)
+    {
+        int exact, normal_ret, range_ret;
+        exact = normal_ret = range_ret = 0;
+        if (next) {
+            range_ret = range_cursor->next(range_cursor.get());
+            /*
+             * If the key exists, position the cursor to the lower key using search near otherwise
+             * use prev().
+             */
+            if (lower_bound.key.length() != 0) {
+                normal_cursor->set_key(normal_cursor.get(), lower_bound.key.c_str());
+                normal_ret = normal_cursor->search_near(normal_cursor.get(), &exact);
+                if (normal_ret == WT_NOTFOUND)
+                    return;
+                if (exact < 0)
+                    normal_ret = normal_cursor->next(normal_cursor.get());
+            } else
+                normal_ret = normal_cursor->next(normal_cursor.get());
+        } else {
+            range_ret = range_cursor->prev(range_cursor.get());
+            /*
+             * If the key exists, position the cursor to the upper key using search near otherwise
+             * use next().
+             */
+            if (upper_bound.key.length() != 0) {
+                normal_cursor->set_key(normal_cursor.get(), upper_bound.key.c_str());
+                normal_ret = normal_cursor->search_near(normal_cursor.get(), &exact);
+                if (normal_ret == WT_NOTFOUND)
+                    return;
+                if (exact > 0)
+                    normal_ret = normal_cursor->prev(normal_cursor.get());
+            } else
+                normal_ret = normal_cursor->prev(normal_cursor.get());
+        }
+
+        if (normal_ret == WT_NOTFOUND)
+            return;
+
+        testutil_assert(normal_ret == range_ret && (normal_ret == 0 || normal_ret == WT_NOTFOUND));
+
+        /* Retrieve the key the cursor is pointing at. */
+        const char *normal_key, *range_key;
+        testutil_check(normal_cursor->get_key(normal_cursor.get(), &normal_key));
+        testutil_check(range_cursor->get_key(range_cursor.get(), &range_key));
+        testutil_assert(strcmp(range_key, normal_key) == 0);
```

Comment:
np: we could stay away from C and avoid using `strcmp` and use [string::compare](https://www.cplusplus.com/reference/string/string/compare/) instead.

Reply:
Hmm currently normal_key and range_key are both char*, should I convert these to std::string instead?

Reply:
You want to keep them as `const char *` otherwise, the WT API won't work. I think it's fine the way you did it but we could also do:
```
std::string(range_key).compare(normal_key)
```
But it was just a suggestion to stay within the C++ realm.

Reply:
Agree that we should stick to c++ methods where we can.